python/ray/serve/metric/client.py

```diff
@@ -0,0 +1,151 @@
+import asyncio
+from typing import Dict, Optional, Tuple
+
+from ray.serve.metric.types import (
+    MetricType,
+    convert_event_type_to_class,
+    MetricMetadata,
+)
+from ray.serve.utils import (retry_actor_failures, retry_actor_failures_async,
+                             _get_logger)
+
+logger = _get_logger()
+
+
+class MetricClient:
+    def __init__(
+            self,
+            metric_exporter_actor,
+            push_interval: float,
+            default_labels: Optional[Dict[str, str]] = None,
+    ):
+        """Initialize a client to push metrics to the exporter actor.
+
+        Args:
+            metric_exporter_actor: The actor to push metrics to.
+            default_labels(dict): The set of labels to apply for all metrics
+                created by this actor. For example, {"source": "worker"}.
+        """
+        self.exporter = metric_exporter_actor
+        self.default_labels = default_labels or dict()
+
+        self.registered_metrics: Dict[str, MetricMetadata] = dict()
+        self.metric_records = []
+
+        assert asyncio.get_event_loop().is_running()
+        self.push_task = asyncio.get_event_loop().create_task(
+            self.push_to_exporter_forever(push_interval))
+        logger.debug("Initialized client")
+
+    @staticmethod
+    def connect_from_serve(default_labels: Optional[Dict[str, str]] = None):
+        """Create the metric client automatically when running inside serve."""
+        from ray.serve.api import _get_master_actor
+
+        master_actor = _get_master_actor()
+        [metric_exporter, push_interval] = retry_actor_failures(
```

Comment:
The push_interval is just a constant now, right? Let's just move the constant here